### PR TITLE
feat: new latency chart and fixed issue with full modal chart

### PR DESCRIPTION
**Pull request by pghildiyal**
# Description

User should be able to select or provide latency percentile for the latency charts of App metrics.
Also in full screen view latency chart is not visible.

Fixes # devtron-labs/devtron#991

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
This has been selected via UI by changing values from dropdown

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/utils.tsx [hunk 1]
```diff
@@ -204,6 +204,60 @@ export function ThroughputSelect(props) {
...
+        placeholder="Latency"
+        value={{ label: props.latency, value: props.latency }}
+        options={[
+            { label: '.999', value: '.999' },
+            { label: '.995', value: '.995' },
+            { label: '.99', value: '.99' },
+            { label: '.95', value: '.95' }
+        ]}
+        onChange={props.handleLatencyChange}
+        styles={{
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/utils.tsx:218]: [Refactoring] style can be extracted out to local variable (as same style is being used in throughput select).
However can be done later.

File: src/components/app/details/appDetails/GraphsModal.tsx [hunk 2]
```diff
@@ -290,6 +312,8 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
                                 </h3>
                                 {this.state.mainChartName === "status" &&
                                     <ThroughputSelect status={this.state.statusCode} handleStatusChange={this.handleStatusChange} />}
+                                {this.state.mainChartName === "latency" &&
+                                    <LatencySelect latency={this.state.selectedLatency} handleStatusChange={this.handleLatencyChange} />}
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/GraphsModal.tsx:316]: [bug]  handleStatusChange -> handleLatencyChange

File: src/components/app/details/appDetails/GraphsModal.tsx [hunk 3]
```diff
@@ -218,6 +220,26 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
         });
     }
 
+    handleLatencyChange(selected): void {
+        let k8sVersion = this.props.k8sVersion;
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/GraphsModal.tsx:224]: [Refactoring] refactoring required as most of things same in handleStatusChange function.  
However can be done later.

File: src/components/app/details/appDetails/GraphsModal.tsx [hunk 4]
```diff
@@ -102,15 +104,15 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
 
         let cpu = getIframeSrc(appInfo, ChartType.Cpu, this.state.calendarInputs, tab, false);
         let ram = getIframeSrc(appInfo, ChartType.Ram, this.state.calendarInputs, tab, false);
-        let latency = getIframeSrc(appInfo, ChartType.Latency, this.state.calendarInputs, tab, false);
+        let latency = getIframeSrc(appInfo, ChartType.Latency, this.state.calendarInputs, tab, false, undefined, this.state.selectedLatency);
         let status2xx = getIframeSrc(appInfo, ChartType.Status, this.state.calendarInputs, tab, false, StatusType.status2xx);
         let status4xx = getIframeSrc(appInfo, ChartType.Status, this.state.calendarInputs, tab, false, StatusType.status4xx);
         let status5xx = getIframeSrc(appInfo, ChartType.Status, this.state.calendarInputs, tab, false, StatusType.status5xx);
-        let status = getIframeSrc(appInfo, ChartType.Status, this.state.calendarInputs, tab, false, StatusType.Throughput);
+        // let status = getIframeSrc(appInfo, ChartType.Status, this.state.calendarInputs, tab, false, StatusType.Throughput);
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/GraphsModal.tsx:111]: delete this line

File: src/components/app/details/appDetails/GraphsModal.tsx [hunk 5]
```diff
@@ -67,13 +68,14 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
             status2xx: "",
             status4xx: "",
             status5xx: "",
-            statusCode: StatusType.Throughput,
+            statusCode: StatusType.status5xx,
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/GraphsModal.tsx:71]: Is this intentional?

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 6]
```diff
@@ -39,6 +39,7 @@ export const AppMetrics: React.FC<{ appName: string, environment, podMap: Map<st
     const { appId, envId } = useParams<AppDetailsPathParams>();
     const [calendarValue, setCalendarValue] = useState('');
     const [statusCode, setStatusCode] = useState<StatusTypes>(StatusType.Throughput);
+    const [latency, setLatency] = useState<number>(.999);
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/AppMetrics.tsx:42]: We should move .999 to some constant (DEFAULT_LATENCY) and use it everywhere instead of hardcoding.

**Review by manish-agrawal-ai — COMMENTED** [review]
check comments

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 7]
```diff
@@ -201,7 +201,7 @@ export const AppMetrics: React.FC<{ appName: string, environment, podMap: Map<st
     }, [calendarValue])
 
     //@ts-ignore
-    if (!datasource.isConfigured || !datasource.isHealthy  || !hostURLConfig || hostURLConfig.value !== window.location.origin ) {
+    if (!datasource.isConfigured || !datasource.isHealthy  /* || !hostURLConfig || hostURLConfig.value !== window.location.origin */ ) {
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/AppMetrics.tsx:204]: revert this change

File: src/components/app/details/appDetails/GraphsModal.tsx (hunk 5 above)
Comment (pghildiyal) [reply src/components/app/details/appDetails/GraphsModal.tsx:71]: yes

File: src/components/app/details/appDetails/AppMetrics.tsx (hunk 6 above)
Comment (pghildiyal) [reply src/components/app/details/appDetails/AppMetrics.tsx:42]: now it is being used at only one place that is here, so no need to move it out

File: src/components/app/details/appDetails/AppMetrics.tsx (hunk 7 above)
Comment (pghildiyal) [reply src/components/app/details/appDetails/AppMetrics.tsx:204]: done

File: src/components/app/details/appDetails/GraphsModal.tsx (hunk 4 above)
Comment (pghildiyal) [reply src/components/app/details/appDetails/GraphsModal.tsx:112]: deleted

File: src/components/app/details/appDetails/utils.tsx [hunk 8]
```diff
@@ -341,6 +363,12 @@ export function addQueryParamToGrafanaURL(url: string, appId: string | number, e
             url += (statusCode.includes("xx")) ? `&var-response_code=` : `&var-response_code=${statusCode}`;
         }
     }
+    if (chartName === 'latency') {
+        if (!isNaN(latency) ) {
+            latency = latency/100;
```
Comment (manish-agrawal-ai) [inline src/components/app/details/appDetails/utils.tsx:368]: should it be - 
latency = (latency/100).toFixed(3);
as (99.9/100) gives 0.9990000000000001
?

Comment (pghildiyal) [reply src/components/app/details/appDetails/utils.tsx:368]: user can give more digits for eg 99.999 which would depend on SLA company follows

**Review by manish-agrawal-ai — APPROVED** [review]
approved

**Review by manish-agrawal-ai — APPROVED** [review]
approved

**Review by manish-agrawal-ai — APPROVED** [review]
approved
